gobblin-runtime/src/main/java/gobblin/runtime/api/JobSpec.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.api;
+
+import java.net.URI;
+import java.util.Optional;
+
+import gobblin.annotation.Alpha;
+
+/**
+ * Defines a Gobblin Job that can be run once, or multiple times. A {@link JobSpec} is
+ * {@link Configurable} so it has an associated {@link Config}, along with other mandatory
+ * properties such as a uri, description, and version. A {@link JobSpec} is
+ * uniquely identified by its uri (containing group and name).
+ *
+ */
+@Alpha
+public interface JobSpec extends Configurable {
```

Comment:
Mostly because it is easier to separate what is the public API on which other components can depend on from the implementation details. Don't feel strongly about interface or class.